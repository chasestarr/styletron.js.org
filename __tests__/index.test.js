import React from "react";
import renderer from "react-test-renderer";
import Button from "../components/snaptest";
import { Provider as StyletronProvider } from "styletron-react";
import { Server } from "styletron-engine-atomic";

test("Button", () => {
  const styletron = new Server();
  const component = renderer.create(
    <StyletronProvider value={styletron}>
      <Button>Text</Button>
    </StyletronProvider>
  );
  const tree = component.toJSON();
  expect(styletron.getCss()).toMatchSnapshot("CSS");
  expect(tree).toMatchSnapshot("HTML");
});
