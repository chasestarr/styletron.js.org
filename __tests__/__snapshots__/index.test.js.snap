// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button: CSS 1`] = `".ae{color:red}"`;

exports[`Button: HTML 1`] = `
<button
  className="ae"
>
  Text
</button>
`;
